rero_ils/modules/local_entities/api.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2023 RERO
+# Copyright (C) 2019-2023 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""API for manipulating local entities."""
+
+from functools import partial
+
+from rero_ils.modules.api import IlsRecord, IlsRecordsIndexer, IlsRecordsSearch
+from rero_ils.modules.fetchers import id_fetcher
+from rero_ils.modules.minters import id_minter
+from rero_ils.modules.providers import Provider
+
+from .dumpers import replace_refs_dumper
+from .extensions import AuthorizedAccessPointExtension
+from .models import LocalEntityIdentifier, LocalEntityMetadata
+
+# provider
+from ..documents.api import DocumentsSearch
+from ..utils import sorted_pids
+
+LocalEntityProvider = type(
+    'LocalEntityProvider',
+    (Provider,),
+    dict(identifier=LocalEntityIdentifier, pid_type='locent')
+)
+# minter
+local_entity_id_minter = partial(id_minter, provider=LocalEntityProvider)
+# fetcher
+local_entity_id_fetcher = partial(id_fetcher, provider=LocalEntityProvider)
+
+
+class LocalEntitiesSearch(IlsRecordsSearch):
+    """Local entities search."""
+
+    class Meta:
+        """Meta class."""
+
+        index = 'local_entities'
+        doc_types = None
+        fields = ('*', )
+        facets = {}
+
+        default_filter = None
+
+
+class LocalEntity(IlsRecord):
```

Comment:
suggestions for next PR's
Use this class as abstract parent class for children specific classes (OrganisationLocalEntity, ConceptLocalEntity, ...) and use a factory pattern to return the correct recordType. Something like 
```python
record = LocalEntity({...})
print(record.__class__.__name__)
> ConceptLocalEntity

record = LocalEntity.get_record[_by_pid](org_locent_pid)
print(record.__class__.__name__)
> OrganisationLocalEntity
```
Using this way, each subclasses could define it's own "authorized_access_point" transformation method.
I used something similar for `Notification` but it's not perfect.

Reply:
why not. We keep in mind the idea 👍